integration/runner/postgres.go

```diff
@@ -133,27 +132,40 @@ func (cc *PostgresClusterController) createAndStartNode(
 			"ALLOW_EMPTY_PASSWORD=yes",
 		}, nodeCreationOpts.additionalEnvs...),
 	}
-
 	cc.nodes = append(cc.nodes, node)
-	node.StartContainer(ctx, t)
-
-	require.NoError(t, nodeStartupRetry.Execute(ctx, func() error {
-		t.Logf("starting db node %v with role: %v", node.Name, node.Role)
-		node.StartContainer(ctx, t)
-		return node.EnsureNodeReadiness(t, nodeCreationOpts.requiredOutput)
-	}))
 
 	return node
 }
 
-// PromoteFollowerNode runs a script that promotes the first follower db node
+func (cc *PostgresClusterController) getNodesConnections(ctx context.Context, t *testing.T) *dbtest.Connection {
+	t.Helper()
+	endpoints := make([]*connection.Endpoint, len(cc.nodes))
+	for i, node := range cc.nodes {
+		endpoints[i] = node.GetContainerConnectionDetails(ctx, t)
+	}
+
+	return dbtest.NewConnection(endpoints...)
+}
+
+func (cc *PostgresClusterController) getLeaderHost(ctx context.Context, t *testing.T) string {
```

Comment:
minor: We should use consistent naming. 
```suggestion
func (cc *PostgresClusterController) getPrimaryHost(ctx context.Context, t *testing.T) string {
```